x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateParse.java

```diff
@@ -63,17 +68,38 @@ public DateParse(
             name = "dateString",
             type = { "keyword", "text" },
             description = "Date expression as a string. If `null` or an empty string, the function returns `null`."
-        ) Expression second
+        ) Expression second,
+        @Param(name = "dateLocale", type = { "keyword", "text" }, description = "The locale to parse with") Expression third,
+        @Param(name = "dateTimezone", type = { "keyword", "text" }, description = "The timezone to parse with") Expression forth
```

Comment:
As discussed, these would fit better in a MapParam. We're currently working on an initiative to add them to some other functions in a standard way.
For context, some other functions already have them (TZ at least). E.g.:

https://github.com/elastic/elasticsearch/blob/f74e8e460c86a54fa051d8cd52d44fbb1c78ea99/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/QueryString.java#L257-L263